src/server/game/Entities/Player/PlayerStorage.cpp

```diff
@@ -6472,9 +6472,15 @@ void Player::_LoadSpells(PreparedQueryResult result)
     if (result)
     {
         do
-            // xinef: checked
-            addSpell((*result)[0].Get<uint32>(), (*result)[1].Get<uint8>(), true);
-        while (result->NextRow());
+        {
+            Field* fields = result->Fetch();
+            uint32 spellId = fields[0].Get<uint32>();
+            uint8 specMask = fields[1].Get<uint8>();
+            addSpell(spellId, specMask, true);
+
+            if (!CheckSkillLearnedBySpell(spellId))
+                removeSpell(spellId, SPEC_MASK_ALL, false);
```

Comment:
```suggestion

            if (CheckSkillLearnedBySpell(spellId))
                addSpell(spellId, specMask, true);
             else
                removeSpell(spellId, SPEC_MASK_ALL, false);
```

I think this is what he meant. We are still only dealing with it in the load, but we dont add if its invalid anyway

Reply:
ah valid